waterlog/src/reducers/leakLitresReducer.js

```diff
@@ -0,0 +1,18 @@
+import { FETCH_LEAK_LITRES } from "../actions/types";
+
+const initialState = {
+  items: [],
+  item: {}
+};
+
+export default function (state = initialState, action) {
+  switch (action.type) {
+    case FETCH_LEAK_LITRES:
+      return {
```

Comment:
Have you tested this? According to the docs it's better to copy the states in the reducer e.g. return Object.assign({}, state, { item: action.payload })

Reply:
Agreed, reason being is that you have to be careful with sending through references to an object instead of a new object. If you don't send through a brand new object reference it messes with reacts change detection and will not re-render the page

Reply:
I have updated my reducers.